File: LuaScriptManager/testcases.lua
fun00 = function () print("function void -> void OK!") end
fun01 = function () print("function void -> int OK!") return 1 end
fun10 = function (i) print("function int -> void OK!") print("Input value:" .. i) end
fun11 = function (i) print("function int -> int OK!") print("Input value:" .. i) return i end
fun20 = function (i,j) print("function (int, int) -> void OK!") print("Input value:" .. i .. ", " .. j) end
fun21 = function (i,j) print("function (int, int) -> int OK!") print("Input value:" .. i .. ", " .. j) return i+j end
fun30 = function (i,j,k) 
	print("function (int, int, int) -> void OK!") 
	print("Input value:" .. i .. ", " .. j .. ", " .. k) 
end

File: LuaScriptManager/testcases.cpp
#include "nflua.h"
#include <string>
#include <fstream>
using std::string;
using std::ifstream;

string readFile(const string& filePath)
{
	ifstream ifs(filePath);
	string data;
	string line;
	while (std::getline(ifs, line))
	{
		data += line + "\n";
	}
	return data;
}


int main(int argc, char** argv)
{
	string scriptData = readFile("testcases.lua");
	NFLua::NFLuaHandler handler(scriptData);
	auto fun00 = handler.getFunc<void>("fun00");
	auto fun01 = handler.getFunc<lua_Number>("fun01");
	auto fun10 = handler.getFunc<void, lua_Number>("fun10");
	auto fun11 = handler.getFunc<lua_Number, lua_Number>("fun11");

	fun00();
	auto r01 = fun01();
	fun10(1);
	auto r11 = fun11(1);
		
	return 0;

}